### PR TITLE
Drop dependency on Python by implementing a small Java SHA256 hasher

### LINKED ISSUE #291: Drop dependency on python for sha256 hashing
Multiple reports:

- https://github.com/bazelbuild/rules_jvm_external/issues/281
- https://stackoverflow.com/questions/58827854/bazel-sha256-checksum/58843439#58843439

It's a major issue on Windows, not so much on *nix platforms.

### PR DESCRIPTION
**Pull request by jin**
Fixes #291 

Many Windows users do not have Python installed (see #291), but that's what we use for sha256 hashing currently (`@bazel_tools//tools/build_defs/hash:sha256.py`). This PR implements a small hasher written in Java.

Also regenerate `maven_install.json` for a couple of examples to test that the generated sha256 is correct.

### CODE REVIEW

File: private/tools/java/rules/jvm/external/Hasher.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    MessageDigest digest = MessageDigest.getInstance("SHA-256");
+    try (BufferedInputStream bufferedInputStream =
+        new BufferedInputStream(new FileInputStream(filename))) {
+      while ((count = bufferedInputStream.read(buffer)) > 0) {
+        digest.update(buffer, 0, count);
+      }
+    }
+    // Convert digest byte array to a hex string.
+    byte[] hashDigest = digest.digest();
+    StringBuffer hexString = new StringBuffer();
```
Comment (artem-zinnatullin) [inline private/tools/java/rules/jvm/external/Hasher.java]: pls don't use `StringBuffer`, all of its methods are `synchronized`, `StringBuilder` is what you want :)

See `StringBuffer` javadoc for details https://docs.oracle.com/javase/10/docs/api/java/lang/StringBuffer.html

File: private/tools/java/rules/jvm/external/Hasher.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/** A tool to compute the sha256 hash of a file. */
+public class Hasher {
+
+  public static void main(String[] args) throws NoSuchAlgorithmException, IOException {
+    // Since this tool is for private usage, just do a simple assertion for the filename argument.
+    assert (args.length == 1) : "Please specify the path of the file to hash.";
```
Comment (artem-zinnatullin) [inline private/tools/java/rules/jvm/external/Hasher.java]: but do you ever expect to run with assertions enabled? `-ea`

Comment (artem-zinnatullin) [reply private/tools/java/rules/jvm/external/Hasher.java]: Resolved :)

Comment (artem-zinnatullin) [reply private/tools/java/rules/jvm/external/Hasher.java]: Or I guess you can just remove the assertion now since you throw exception otherwise

Comment (jin) [reply private/tools/java/rules/jvm/external/Hasher.java]: done, thanks.

File: private/tools/java/rules/jvm/external/Hasher.java (hunk 1 above)
Comment (jin) [reply private/tools/java/rules/jvm/external/Hasher.java]: done.

**Pull request merged**